src/app/domain/core/validators/auth.validators.ts

```diff
@@ -65,3 +65,25 @@ export const authenticateRequest = () => {
 };
 
 export type LoginInput = yup.InferType<typeof loginValidationSchema>;
+
+
+// Parameters needed for password reset
+export const passwordResetValidationSchema = yup.object({
+  currentPassword: yup.string().required("Password is required"),
+
+  newPassword: yup.string().required("Password is required"),
+
+  confirmNewPassword: yup
+    .string()
+    .oneOf([
+      yup.ref("new_password"),
```

Comment:
If you want the`newPassword` and `confirmNewPassword` to match, then you need to change the `yup.ref('new_password')` to be `yup.ref('newPassword')` instead. 

This is because yup will attempt to find any key of `new_password` in the `passwordResetValidationSchema` and if its not found, the newPassword and confirmNewPassword will never be same.

Reply:
Done